packages/iTwinUI-react/src/core/utils/components/VirtualScroll.tsx

```diff
@@ -215,8 +184,7 @@ export const useVirtualization = (props: VirtualScrollProps) => {
     return arr;
   }, [itemsLength, itemRenderer, bufferSize, startNode, visibleNodeCount]);
 
-  // Get child height when children available
-  React.useLayoutEffect(() => {
+  const updateChildHeight = React.useCallback(() => {
```

Comment:
**[COMMENT]** The main change I made was to abstract the setting of `childHeight` (old lines 220-242). Then `updateChildHeight()` can be called in `React.useLayoutEffect()` and in `onResize()`. That way when the resize happens (`display` goes from `display: none` to `display: block`), the `childHeight` is updated which allows all us to visit all the items.